gobblin-core/src/main/java/gobblin/publisher/BaseDataPublisher.java

```diff
@@ -133,13 +143,68 @@ public BaseDataPublisher(State state) throws IOException {
     this.parallelRunnerCloser = Closer.create();
   }
 
+  // Build the list of metadataMergers for each branch. Return null if none are configured at all
+  private List<MetadataMerger<String>> mergersForEachBranch() {
+    List<MetadataMerger<String>> metadataMergers = new ArrayList<>();
+    boolean mergersConfigured = false;
+
+    for (int i = 0; i < numBranches; i++) {
+      if (shouldPublishMetadataForBranch(i)) {
+        metadataMergers.add(buildPublisherForBranch(i));
+        mergersConfigured = true;
+      } else {
+        metadataMergers.add(null);
+      }
+    }
+
+    return mergersConfigured ? metadataMergers : null;
```

Comment:
I originally had it that way but decided that a check for (mergers == null) later on makes more sense than mergers.size() == 0. Thoughts? I don't feel super strongly either way.